smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpClientHelper.java

```diff
@@ -34,7 +35,11 @@ static AmqpClient createClient(AmqpConnector connector, AmqpConnectorCommonConfi
 
     static AmqpClient createClientFromClientOptionsBean(Vertx vertx, Instance<AmqpClientOptions> instance,
             String optionsBeanName) {
-        Instance<AmqpClientOptions> options = instance.select(NamedLiteral.of(optionsBeanName));
+        Instance<AmqpClientOptions> options = instance.select(Identifier.Literal.of(optionsBeanName));
+        if (options.isUnsatisfied()) {
+            // this `if` block should be removed when support for the `@Named` annotation is removed
+            options = instance.select(NamedLiteral.of(optionsBeanName));
```

Comment:
Should we add a warning for the user to invite him to switch to the new annotation?